### PR TITLE
Close focused menu on pressing escape

### PR DESCRIPTION
**Pull request by peppy**

### CODE REVIEW

File: osu.Framework/Graphics/UserInterface/Menu.cs [hunk 1]
```diff
@@ -465,6 +466,18 @@ private void menuItemHovered(DrawableMenuItem item)
             }
         }
 
+        protected override bool OnKeyDown(InputState state, KeyDownEventArgs args)
+        {
+            switch (args.Key)
+            {
+                case Key.Escape:
+                    Close();
```
Comment (smoogipooo) [inline osu.Framework/Graphics/UserInterface/Menu.cs]: Have you tested that this works with nested menus?

Comment (smoogipooo) [reply osu.Framework/Graphics/UserInterface/Menu.cs]: i.e. this might have to be `closeAll()`.

Comment (peppy) [reply osu.Framework/Graphics/UserInterface/Menu.cs]: It should only close one level of menu IMO, not all.

Comment (smoogipooo) [reply osu.Framework/Graphics/UserInterface/Menu.cs]: Ah good point.

**Pull request merged**